src/resources/Sources/SourcesInterfaces.ts

```diff
@@ -0,0 +1,606 @@
+    sourceVisibility?: SourceVisibility;
+    urlFilters?: UrlFilter[];
+}
+
+export interface ConfigurationError {
+    errorCode?: string;
+    message?: string;
+}
+
+export interface SourceInformation {
+    collectionId?: number;
+    collectionName?: string;
+    documentsTotalSize?: number;
+    lastOperation?: SourceLastOperation;
+    nextOperation?: NextOperation;
+    numberOfDocuments?: number;
+    rebuildRequired?: boolean;
+    sourceId?: number;
+    sourceName?: string;
+    sourceStatus?: SourceStatus;
+}
+
+export interface MappingModel {
+    content?: string;
+    extractionMethod?: string;
+    fieldName?: string;
+    fieldTypeHint?: FieldTypes;
+    id?: string;
+    kind?: 'COMMON' | 'MAPPING';
+    type?: string;
+}
+
+export interface DocumentPermissionModel {
+    lastUpdatedDate?: number;
+    permissionLevels?: PermissionLevelModel[];
+    state?: 'UNKNOWN' | 'VALID' | 'PENDING' | 'INCOMPLETE' | 'IN_ERROR' | 'WARNING';
+}
+
+export interface SourceExtensionModel {
+    actionOnError?: 'REJECT_DOCUMENT' | 'SKIP_EXTENSION';
+    condition?: string;
+    extensionId?: string;
+    itemType?: string;
+    parameters?: any;
+    versionId?: string;
+}
+
+export interface UrlFilter {
+    filter?: string;
+    filterType?: 'WILDCARD' | 'REGEX';
+    includeFilter?: boolean;
+}
+
+export interface SourceLastOperation {
+    errorCode?: string;
+    id?: string;
+    initialBuild?: boolean;
+    numberOfDocuments?: number;
+    operationType?: OperationType;
+    result?: string;
+    timestamp?: number;
+}
+
+export interface NextOperation {
+    operationType?: OperationType;
+    timestamp?: number;
+}
+
+export interface SourceStatus {
+    allowedOperations?: ActivityOperation[];
+    initialBuild?: boolean;
+    numberOfDocuments?: number;
+    numberOfProcessedDocuments?: number;
+    pausedOnErrorCode?: string;
+    refreshType?: OperationType;
+    timestamp?: number;
+    type?: SourceStatusType;
+}
+
+export interface PermissionLevelModel {
+    name?: string;
+    permissionSets?: PermissionSetModel[];
+}
+
+export interface PermissionSetModel {
+    anonymousAllowed?: boolean;
+    name?: string;
+    permissions?: SinglePermissionModel[];
+}
+
+export interface SinglePermissionModel {
+    additionalInfo?: any;
+    allowed?: boolean;
+    identity?: string;
+    identityType?: 'USER' | 'GROUP' | 'VIRTUAL_GROUP' | 'UNKNOWN';
+    lastUpdateDate?: number;
+    lastUpdateErrorDetail?: string;
+    lastUpdateResult?:
+        | 'NONE'
+        | 'SUCCESS'
+        | 'ACCESS_DENIED'
+        | 'TIMED_OUT'
+        | 'ENTITY_IS_INVALID'
+        | 'ENTITY_IS_UNAVAILABLE'
+        | 'SECURITY_PROVIDER_IS_UNAVAILABLE'
+        | 'SECURITY_PROVIDER_IS_UNREACHABLE'
+        | 'SECURITY_PROVIDER_IS_NOT_READY'
+        | 'UNEXPECTED_ERROR';
+    securityProvider?: string;
+    state?: 'UNKNOWN' | 'UP_TO_DATE' | 'NOT_UPDATED' | 'OUT_OF_DATE' | 'IN_ERROR' | 'DISABLED';
+}
+
+export enum OperationType {

```

Comment:
Same here, we'll have to move all these in the Enums.ts file once my PR is merged